src/main/java/com/google/sps/servlets/BlobServlet.java

```diff
@@ -33,59 +41,203 @@
   public void doPost(HttpServletRequest request, HttpServletResponse response)
```

Comment:
This looks like one very large function. I'd suggest to break it up into smaller pieces to make it easier to read and follow.

Reply:
Noted